cli/tests/test_advance_branches.rs

```diff
@@ -40,9 +42,37 @@ fn set_advance_branches_overrides(
     )
 }
 
+// Runs a command in the specified test environment and workspace path that
+// describes the current commit with `commit_message` and creates a new commit
+// on top of it.
+type CommitFn = fn(env: &TestEnvironment, workspace_path: &Path, commit_message: &str);
+
+// Implements CommitFn using the `jj commit` command.
+fn commit_cmd(env: &TestEnvironment, workspace_path: &Path, commit_message: &str) {
+    env.jj_cmd_ok(workspace_path, &["commit", "-m", commit_message]);
+}
+
+macro_rules! parameterized_tests{
+    ($($test_name:ident: $case_fn:ident($commit_fn:ident),)*) => {
+    $(
+        #[test]
+        fn $test_name() {
+            $case_fn($commit_fn);
+        }
+    )*
+    }
+}
```

Comment:
I think this will make harder for us to maintain the tests.

Since `jj commit` is deprecated (and is practically the same as `jj new`), I don't think we need to implement the advance-branches feature in `jj commit`. That will simplify the testing scenario.

Reply:
I've been using `commit` almost exclusively. If `new` and `commit` diverge in such a way that `commit` will, without warning, leave my branch behind, that will be very frustrating.

Reply:
Oops, maybe I confused it with the `checkout` deprecation. I also use `jj ci` often.

Anyway, I prefer tests without too much meta programming. If needed, `test_case()` macro could be used, but parameterized tests generally don't work well with insta.

Reply:
Thanks, I didn't know we had `test_case`. I'm going to replace the custom macro with that. insta is cooperating as long as we wrap the expectations in allow_duplicates. I think it's worth parameterizing here since the test expectations are the same.